src/coreclr/pal/src/init/pal.cpp

```diff
@@ -76,16 +76,12 @@ int CacheLineSize;
 #include <sys/param.h>
 #include <sys/sysctl.h>
 #include <kvm.h>
-#elif defined(__sun)
-#ifndef _KERNEL
-#define _KERNEL
-#define UNDEF_KERNEL
-#endif
-#include <sys/procfs.h>
-#ifdef UNDEF_KERNEL
-#undef _KERNEL
```

Comment:
There are three places where we had to use this workaround for a reason (I think some missing definition or behavior difference):

```sh
$ git grep -n "undef _KERNEL"
src/coreclr/pal/src/init/pal.cpp:86:#undef _KERNEL
src/coreclr/pal/src/thread/thread.cpp:45:#undef _KERNEL
src/native/libs/System.Native/pal_io.c:69:#undef _KERNEL
```

Maybe we should revisit them as well if you are sure that `_KERNEL` is not needed now?

Please make sure cross-build scenario is working with this change, otherwise it gets very complicated as we don't have the illumos SDK.

Reply:
The other locations should have the same fix.  All are due to the mistake of using <sys/procfs.h> where just <procfs.h> should be used.  Would it be easier to have that change in a separate PR and fix all three places?

Reply:
Sounds good. Note that I'm not a maintainer. So I'm just giving what "I think" makes sense, which may or may not be the final word when maintainers will review these changes. :)

Reply:
Right, also related to #44417